rosflight/src/rosflight_io.cpp

```diff
@@ -616,25 +616,39 @@ void rosflightIO::handle_small_mag_msg(const mavlink_message_t &msg)
   mag_pub_.publish(mag_msg);
 }
 
-void rosflightIO::handle_small_sonar(const mavlink_message_t &msg)
+void rosflightIO::handle_small_range(const mavlink_message_t &msg)
 {
-  mavlink_small_sonar_t sonar;
-  mavlink_msg_small_sonar_decode(&msg, &sonar);
+  mavlink_small_range_t mRange;
+  mavlink_msg_small_range_decode(&msg, &mRange);
 
   sensor_msgs::Range alt_msg;
   alt_msg.header.stamp = ros::Time::now();
-  alt_msg.max_range = sonar.max_range;
-  alt_msg.min_range = sonar.min_range;
-  alt_msg.range = sonar.range;
+  alt_msg.max_range = mRange.max_range;
+  alt_msg.min_range = mRange.min_range;
+  alt_msg.range = mRange.range;
 
-  alt_msg.radiation_type = sensor_msgs::Range::ULTRASOUND;
-  alt_msg.field_of_view = 1.0472;  // approx 60 deg
+  if (mRange.type == ROSFLIGHT_RANGE_SONAR)
```

Comment:
Would you mind changing this to switch block? It's just a little more explicit and would be easier to expand if we add more sensor types in the future. Just leave an empty default case to avoid compiler warnings